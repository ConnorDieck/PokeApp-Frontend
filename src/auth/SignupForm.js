import React, { useState } from "react";
import { Link, useNavigate } from "react-router-dom";
import { Grid, FormControl, TextField, Button, Paper, Avatar, Typography } from "@mui/material";
import ExitToAppOutlinedIcon from "@mui/icons-material/ExitToAppOutlined";

const SignupForm = ({ register }) => {
	let INITIAL_STATE = {
		username : "",
		password : ""
	};

	const [ fData, setFormData ] = useState(INITIAL_STATE);
	const navigate = useNavigate();

	const handleChange = evt => {
		const { name, value } = evt.target;
		setFormData(fData => ({
			...fData,
			[name] : value
		}));
	};

	// when submitted, runs register function the current user and token to state
	const handleSubmit = async evt => {
		evt.preventDefault();
		let res = await register(fData);
		console.log(res);
		setFormData(INITIAL_STATE);
		if (res.success) {
			navigate("/favorite");
		} else {
			alert(res.err);
		}
	};

	const paperStyle = {
		padding : 20,
		height  : "auto",
		width   : 280,
		margin  : "20px auto"
	};

	const avatarStyle = {
		backgroundColor : "#22d46c"
	};

	const btnStyle = {
		margin : "8px 0"
	};

	return (
		<Grid>
			<Paper elevation={10} style={paperStyle}>
				<Grid align="center">
					<Avatar style={avatarStyle}>
						<ExitToAppOutlinedIcon />
					</Avatar>
				</Grid>
				<h2>Sign Up</h2>
				<form className="auth-form" onSubmit={handleSubmit}>
					<FormControl>
						<TextField //
							label="Username"
							id="register_username"
							name="username"
							value={fData.username}
							onChange={handleChange}
							placeholder="Username"
							className="styled-input"
							fullWidth
							required
						/>

						<TextField //
							label="Password"
							id="register_password"
							name="password"
							type="password"
							value={fData.password}
							onChange={handleChange}
							placeholder="Password"
							className="styled-input"
							fullWidth
							required
						/>

						<Button variant="contained" color="secondary" style={btnStyle} type="submit">
							Sign Up
						</Button>
					</FormControl>
				</form>
				<Typography>
					Already have an account?
					<Link to="/login">Log In</Link>
				</Typography>
			</Paper>
		</Grid>
	);
};

export default SignupForm;
