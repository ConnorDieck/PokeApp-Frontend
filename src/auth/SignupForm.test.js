import React from "react";
import { render } from "@testing-library/react";
import SignupForm from "./SignupForm";
import { MemoryRouter } from "react-router-dom";

//smoke test
it("renders without crashing", function() {
	render(
		<MemoryRouter>
			<SignupForm />
		</MemoryRouter>
	);
});

// snapshot test
test("matches snapshot", function() {
	const { asFragment } = render(
		<MemoryRouter>
			<SignupForm />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});
