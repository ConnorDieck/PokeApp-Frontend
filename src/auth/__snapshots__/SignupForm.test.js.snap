// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root css-vj1n65-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation10 css-14epxft-MuiPaper-root"
      style="padding: 20px; height: auto; width: 280px; margin: 20px auto;"
    >
      <div
        align="center"
        class="MuiGrid-root css-vj1n65-MuiGrid-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
          style="background-color: rgb(34, 212, 108);"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExitToAppOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10.09 15.59 11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </div>
      </div>
      <h2>
        Sign Up
      </h2>
      <form
        class="auth-form"
      >
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root styled-input css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="register_username"
              id="register_username-label"
            >
              Username
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="register_username"
                name="username"
                placeholder="Username"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Username *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root styled-input css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="register_password"
              id="register_password-label"
            >
              Password
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="register_password"
                name="password"
                placeholder="Password"
                required=""
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Password *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-zcbmsk-MuiButtonBase-root-MuiButton-root"
            style="margin: 8px 0px;"
            tabindex="0"
            type="submit"
          >
            Sign Up
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Already have an account?
        <a
          href="/login"
        >
          Log In
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
